content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,213 @@
+
+```ts
+import { context } from '@opentelemetry/api'
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { Resource } from '@opentelemetry/resources'
+import {
+  BasicTracerProvider,
+  SimpleSpanProcessor,
+  ConsoleSpanExporter,
+} from '@opentelemetry/sdk-trace-base'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { PrismaClient } from '@prisma/client'
+
+const contextManager = new AsyncHooksContextManager().enable()
+
+context.setGlobalContextManager(contextManager)
+
+const provider = new BasicTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+
+// Console exporter
+const consoleExporter = new ConsoleSpanExporter()
+provider.addSpanProcessor(new SimpleSpanProcessor(consoleExporter))
+provider.register()
+
+// Register Prisma Instrumentation
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  const email = `user.${Date.now()}@prisma.io`
+
+  await prisma.user.create({
+    data: {
+      email: email,
+    },
+  })
+}
+
+main()
+...
+```
+
+## Send tracing to Jaeger
+
+To send tracing to Jaeger, add the following to you application code:
+
+```ts
+import { context } from '@opentelemetry/api'
+import { AsyncHooksContextManager } from '@opentelemetry/context-async-hooks'
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http'
+import { registerInstrumentations } from '@opentelemetry/instrumentation'
+import { Resource } from '@opentelemetry/resources'
+import {
+  BasicTracerProvider,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base'
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions'
+import { PrismaInstrumentation } from '@prisma/instrumentation'
+import { PrismaClient } from '@prisma/client'
+
+const contextManager = new AsyncHooksContextManager().enable()
+
+context.setGlobalContextManager(contextManager)
+
+const provider = new BasicTracerProvider({
+  resource: new Resource({
+    [SemanticResourceAttributes.SERVICE_NAME]: 'test-tracing-service',
+    [SemanticResourceAttributes.SERVICE_VERSION]: '1.0.0',
+  }),
+})
+
+// Jager exporter
+const otlpTraceExporter = new OTLPTraceExporter()
+provider.addSpanProcessor(new SimpleSpanProcessor(otlpTraceExporter))
+provider.register()
+
+// Register Prisma Instrumentation
+registerInstrumentations({
+  instrumentations: [new PrismaInstrumentation()],
+})
+
+async function main() {
+  const prisma = new PrismaClient()
+
+  const email = `user.${Date.now()}@prisma.io`
+
+  await prisma.user.create({
+    data: {
+      email: email,
+    },
+  })
+}
+
+main()
+...
+```
```

Comment:
May want to call out that Honeycomb, Datadog, etc. also work. I'm up for testing these during QA.

Reply:
We do say that we support any OpenTelemetry tracing system. There's no particular reason we should just mention Jaeger by name, though, so I'll add these others. If our testing shows they don't work, we can remove them.

Reply:
Cool, I think for SEO it would be nice too.